test/pylib/manager_client.py

```diff
@@ -239,3 +242,31 @@ async def get_host_id(self, server_id: ServerNum) -> HostID:
         except Exception as exc:
             raise Exception(f"Failed to get local host id address for server {server_id}") from exc
         return HostID(host_id)
+
+    async def server_sees_others(self, server_id: ServerNum, count: int, deadline: int = 45):
+        """Wait till a server sees a minimum given count of other servers"""
+        if count < 1:
+            return
+        server_ip = await self.get_host_ip(server_id)
+        async def _sees_min_others():
+            alive_nodes = await self.api.get_alive_endpoints(server_ip)
+            if len(alive_nodes) > count:
+                return True
+        await wait_for(_sees_min_others, time() + deadline, period=.1)
+
+    async def server_sees_other_server(self, server_ip: IPAddress, other_ip: IPAddress,
+                                       deadline: int = 45):
+        """Wait till a server sees another specific server IP as alive"""
+        async def _sees_another_server():
+            alive_nodes = await self.api.get_alive_endpoints(server_ip)
+            if other_ip in alive_nodes:
+                return True
+        await wait_for(_sees_another_server, time() + deadline, period=.1)
+
+    async def server_not_sees_other_server(self, server_ip: IPAddress, other_ip: IPAddress):
+        """Wait till a server sees another specific server IP as dead"""
+        async def _not_sees_another_server():
+            alive_nodes = await self.api.get_alive_endpoints(server_ip)
+            if not other_ip in alive_nodes:
+                return True
+        await wait_for(_not_sees_another_server, time() + 45, period=.1)
```

Comment:
Forgot about parameterizing deadline here

Reply:
Fixed in https://github.com/scylladb/scylladb/commit/1b146e7296dab3a3d9df866582c92340727b7070